src/webcore/serialization.rs

```diff
@@ -214,13 +215,54 @@ impl SerializedUntaggedArray {
     }
 }
 
-pub struct ObjectDeserializer< 'a > {
-    key_slice: &'a [SerializedUntaggedString],
-    value_slice: &'a [SerializedValue< 'a >],
+/// Owns some memory allocated by FFI, and will deallocate it on drop.
+///
+/// `T` must not be zero-sized.
+struct OwnedFfiSlice< T > {
+    ptr: *const T,
+    length: usize,
+}
+
+impl< T > OwnedFfiSlice< T > {
+    unsafe fn new(ptr: *const T, length: usize) -> Self {
+        assert_ne!( mem::size_of::< T >(), 0 );
+        OwnedFfiSlice { ptr, length }
+    }
+}
+
+impl< T > Deref for OwnedFfiSlice< T > {
+    type Target = [ T ];
+
+    fn deref( &self ) -> & [ T ] {
+        assert_ne!( mem::size_of::< T >(), 0 );
+        unsafe { slice::from_raw_parts ( self.ptr, self.length ) }
+    }
+}
+
+impl< T > Drop for OwnedFfiSlice< T >  {
+    fn drop( &mut self ) {
+        assert_ne!( mem::size_of::< T >(), 0 );
+        unsafe {
+            ffi::dealloc( self.ptr as *mut u8, self.length * mem::size_of::< T >() );
+        }
+    }
+}
+
+pub struct ObjectDeserializer {
+    key_slice: OwnedFfiSlice < SerializedUntaggedString >,
+    value_slice: OwnedFfiSlice < SerializedValue< 'static > >,
```

Comment:
I couldn't find any reason why using `'static` here would cause a safety risk, but it seems like it could be wrong in some way since the lifetime parameter does exist there. Since this makes `ObjectDeserializer` own its contents and drop it on drop, it didn't feel right to use any external lifetime parameter here.